crates/pathfinder/src/consensus/inner/proposal_error.rs

```diff
@@ -0,0 +1,71 @@
+//! Error types for proposal handling.
+
+use thiserror::Error;
```

Comment:
I'd suggest _not_ importing `thiserror::Error` and using the qualified name in `#[derive()]`.

I think `Error` is too generic a name here, and using the qualified name in the derive makes things much more clear. That's why we usually don't import `anyhow::Error` either and use it with its qualified name.

Reply:
+1